cmake/add_pstore.cmake

```diff
@@ -266,6 +266,47 @@ function (add_pstore_executable target_name)
 endfunction(add_pstore_executable)
 
 
+############################
+# add_pstore_install_targets
+############################
+
+function(add_pstore_install_targets target)
+  cmake_parse_arguments(ARG "" "COMPONENT;PREFIX" "DEPENDS" ${ARGN})
+  if(ARG_COMPONENT)
+    set(component_option -DCMAKE_INSTALL_COMPONENT="${ARG_COMPONENT}")
+  endif()
+  if(ARG_PREFIX)
+    set(prefix_option -DCMAKE_INSTALL_PREFIX="${ARG_PREFIX}")
+  endif()
+
+  add_custom_target(${target}
+                    DEPENDS ${ARG_DEPENDS}
+                    COMMAND "${CMAKE_COMMAND}"
+                            ${component_option}
+                            ${prefix_option}
+                            -P "${CMAKE_BINARY_DIR}/cmake_install.cmake"
+                    USES_TERMINAL)
+endfunction()
+
+
+#################
+# add_pstore_tool
+#################
+
+function (add_pstore_tool name)
+
+    add_pstore_executable(${name} ${ARGN})
```

Comment:
Is there a reason why the body of add_pstore_executable isn’t here? Is there a need to keep the two functions separate?

Reply:
The idea behind 'install-pstore' is to allow the 'installation' of just the pstore tools. It does not install the pstore examples. That is the reason to keep both functions 'add_pstore_executable' and 'add_pstore_tool'.

May be:
1) Include the pstore examples in the 'install-pstore' target
or
2) Add 'install-pstore-examples' target to install the examples

Reply:
The logic behind keeping the two functions makes sense. However, their names don't really give any suggestion that that's the difference. How about something like add_pstore_installed_tool() and add_pstore_tool()?

I don't think it's necessary to install the example executables. In the same vein, it isn't necessary to install pstore-broker-poker or pstore-hamt-test since these executables are just used by the system tests.

Reply:
I will remove 'pstore-broker-poker' and 'pstore-hamt-test' from the tools being installed.

Reply:
I followed the Clang and LLVM convention to name those 2 functions: add_pstore_executable and add_pstore_tool.
They use: add_clang/llvm_executable' and 'add_clang/llvm_tool.

Reply:
I see: that's a strong precedent to follow. I like to think that we can do better than LLVM though! I'd like to see something – perhaps just some comments – that explains the difference between an "executable" and a "tool".

Reply:
I will add some comments explaining the difference.